File: domain/src/main/kotlin/com/kyovo/domain/port/primary/BookingUseCase.kt
package com.kyovo.domain.port.primary

import com.kyovo.domain.model.booking.*
import com.kyovo.domain.model.user.UserId

interface BookingUseCase
{
    fun findAll(): List<Booking>

    fun findById(id: BookingId): Booking?

    fun findByUserId(userId: UserId): List<Booking>

    fun create(newBooking: NewBooking): Booking

    fun cancel(bookingId: BookingId, cancelledBy: UserId, isAdmin: Boolean, reason: BookingStatusReason?): Booking

    fun findStatusHistory(bookingId: BookingId): List<BookingStatusHistory>

    fun validate(bookingId: BookingId, adminId: UserId): Booking
}

File: frontend/src/domain/port/primary/BookingUseCase.ts
import type { Booking, NewBooking } from '@/domain/model/Booking'

export interface BookingUseCase {
  findAll(): Promise<Booking[]>
  findByUser(userId: string): Promise<Booking[]>
  create(booking: NewBooking): Promise<Booking>
  cancel(id: string, userId: string, reason?: string): Promise<void>
}
